# Python del 关键字

> 原文:[https://codescracker.com/python/python-del-keyword.htm](https://codescracker.com/python/python-del-keyword.htm)

Python 中的 **del** 关键字是在我们需要删除一个对象时使用的。例如:

```
class CodesCracker:
    Name = "William"
    Age = "20"

print("An object 'CodesCracker' is created.")
del CodesCracker
print("The object 'CodesCracker' is now deleted.")

a = 10
print("\nAn object 'a' is created.")
del a
print("The object 'a' is now deleted")

b = [12, 324, 54, 564]
print("\nAn object 'b' is created.")
del b
print("The object 'b' is now deleted")

c = "Python Programming"
print("\nAn object 'c' is created.")
del c
print("The object 'c' is now deleted")
```

下面给出的快照显示了上述程序产生的输出，演示了 Python 中的 **del** 关键字:

![python del keyword](../Images/8827737924365487c58455347f97ac60.png)

**注意-**Python 中的一切都是对象，比如变量、列表、元组、列表的一部分等。

问题是，如果我们试图访问一个在同一个程序中使用 **del** 关键字、 删除的对象，会发生什么。让我们用下面给出的程序来找出答案:

```
x = "Python Programming"
print("The value of 'x' is:", x)

del x

print("\nThe value of 'x' is: ", x)
```

该程序产生的输出显示在下面给出的快照中:

![python del keyword example](../Images/77f10fbf1d5c3cc060134930b419fe4f.png)

说变量 **x** 没有定义。因为，使用 **del** 关键字，这个变量在最后一个语句之前 被删除。因此出现了一个名为 **NameError** 的异常。

使用 **del** 关键字，我们还可以从 iterable like 列表中删除特定的条目。例如:

```
x = [10, 45, 78, 90]
del x[1]
print(x)
```

使用`del x[1]`，索引号为 1 的元素将被删除，即 45。因此，输出是:

```
[10, 78, 90]
```

类似地，一个类的属性也可以使用 **del** 关键字来删除，以这种方式:

```
class CodesCracker:
    Name = "William"
    Age = "20"

del CodesCracker.Name
```

现在 **CodesCracker** 类的 **Name** 属性被删除。关键字 *del* 也可以被 用来从一个 iterable 中删除多个条目。例如:

```
x = [10, 45, 78, 90, 12, 23, 56, 67]
del x[1:5]
print(x)
```

在代码`x[1:5]`中，包含了 1，而排除了 5。这两个是列表的索引号。因此, 的输出是:

```
[10, 23, 56, 67]
```

[Python 在线测试](/exam/showtest.php?subid=10)

* * *

* * *